include/crocoddyl/core/optctrl/shooting.hxx

```diff
@@ -231,6 +229,34 @@ std::vector<typename MathBaseTpl<Scalar>::VectorXs> ShootingProblemTpl<Scalar>::
   return xs;
 }
 
+template <typename Scalar>
+void ShootingProblemTpl<Scalar>::quasiStatic(const std::vector<VectorXs>& us, std::vector<VectorXs>& xs) {
+  if (xs.size() != T_) {
+    throw_pretty("Invalid argument: "
+                 << "xs has wrong dimension (it should be " + std::to_string(T_) + ")");
+  }
+  if (us.size() != T_) {
+    throw_pretty("Invalid argument: "
+                 << "us has wrong dimension (it should be " + std::to_string(T_) + ")");
+  }
+
+#ifdef CROCODDYL_WITH_MULTITHREADING
+#pragma omp parallel for
+#endif
+  for (std::size_t i = 0; i < T_; ++i) {
+    running_models_[i]->quasiStatic(running_datas_[i], xs[i], us[i]);
```

Comment:
The API for quasiStatic is defined as quasiStatic(data, u, x)

Reply:
Good catch! I have solved this issue and added unittest of all main shooting problem functions